modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rule/FilterConverterRule.java

```diff
@@ -37,14 +42,25 @@
 
     /** */
     public FilterConverterRule() {
-        super(LogicalFilter.class);
+        super(LogicalFilter.class, "FilterConverterRule");
     }
 
     /** {@inheritDoc} */
     @Override protected PhysicalNode convert(RelOptPlanner planner, RelMetadataQuery mq, LogicalFilter rel) {
         RelOptCluster cluster = rel.getCluster();
-        RelTraitSet traits = cluster.traitSetOf(IgniteConvention.INSTANCE);
-        RelNode input = convert(rel.getInput(), traits);
+        RelTraitSet traits = rel.getTraitSet().replace(IgniteConvention.INSTANCE);
+        RelTraitSet inTraits = rel.getTraitSet().replace(IgniteConvention.INSTANCE);
+        RelNode input = rel.getInput();
+
+        Set<CorrelationId> corrIds = RexUtils.extractCorrelationIds(rel.getCondition());
+
+        if (!corrIds.isEmpty()) {
+            inTraits = rel.getTraitSet().replace(RewindabilityTrait.REWINDABLE);
```

Comment:
inTraits are unused in this scope.

Reply:
Not catch your point.
We change the traits of the input node in case filter contains correlated expression.